spyder/widgets/ipythonconsole/shell.py

```diff
@@ -87,6 +86,14 @@ def set_cwd(self, dirname):
         else:
             self.silent_execute(code)
 
+    def set_color_scheme(self, color_scheme):
+        """Set color scheme of the shell."""
+        self.style_sheet = create_qss_style()
+        self.syntax_style = get_syntax_style(name=color_scheme)
```

Comment:
Sorry, this is wrong. The right thing to do is not to `_syntax_style_changed` but some parts of it.